rfcs/0035-create-connection.md

```diff
@@ -0,0 +1,143 @@
+---
+feature: Create Connection
+authors:
+  - "Bohan Zhang"
+start_date: "2023/01/15"
+---
+
+# RFC: Create Connection
+
+## Motivation
+
+A connection is used to describe how to connect to an external system that users want to read data from. Once created, a connection is reusable across `CREATE SOURCE` and `CREATE SINK` statements. This RFC proposes a new `CREATE CONNECTION` statement to create a connection.
```

Comment:
Any stronger motivation? I mean, like, beside the convenience it brings, is there anything that was impossible but become doable after it?

Reply:
there is no proper abstraction other than `connection` to connect private link with the existing reader 🤣 
@wyhyhyhyh has asked for this feature in the last Q

Reply:
Emmm, they ("private link" and "connection") look independent with each other. "Connection" here provides not only the network access to VPC but also the very credentials it needs to connect to a specific data source/sink.

I mean, since you are proposing to allow users to write AWS VPC credentials along with the Kafka/Kinis properties e.g.

```sql
CREATE CONNECTION demo_connection
  with (
    connection_type = 'kinesis',
    aws.region='user_test_topic',
    endpoint='172.10.1.1:9090,172.10.1.2:9090',
    aws.credentials.role.arn='arn:aws-cn:iam::602389639824:role/demo_role',
    aws.credentials.role.external_id='demo_external_id',
   ...
  )
```

Then, why not just allow users to write AWS VPC credentials within `CREATE SOURCE`:

```sql
CREATE SOURCE demo_source
  with (
    connector_type = 'kinesis',
    aws.region='user_test_topic',
    endpoint='172.10.1.1:9090,172.10.1.2:9090',
    aws.credentials.role.arn='arn:aws-cn:iam::602389639824:role/demo_role',
    aws.credentials.role.external_id='demo_external_id',
   ...
  )
```

Disclaimer: I am not arguing this syntax is better. I am saying the proposed solution seems not to be an exact answer to the problem/motivation.